pandas/core/internals/construction.py

```diff
@@ -349,14 +349,15 @@ def ndarray_to_mgr(
         if dtype is None and is_object_dtype(values.dtype):
             arrays = [
                 ensure_wrapped_if_datetimelike(
-                    maybe_infer_to_datetimelike(values[:, i])
+                    maybe_infer_to_datetimelike(values[:, i].copy())
                 )
                 for i in range(values.shape[1])
             ]
         else:
             if is_datetime_or_timedelta_dtype(values.dtype):
                 values = ensure_wrapped_if_datetimelike(values)
-            arrays = [values[:, i] for i in range(values.shape[1])]
+            # copy the array to ensure contiguous memory
+            arrays = [values[:, i].copy() for i in range(values.shape[1])]
```

Comment:
could do the copy+comment once on the next line instead of both here and L352

Reply:
I think that would require an additional list comprehension over the arrays?